services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/TopicCleanupService.java

```diff
@@ -401,4 +446,40 @@ public static List<PubSubTopic> extractVersionTopicsToCleanup(
         })
         .collect(Collectors.toList());
   }
+
+  private List<PubSubTopic> collectDanglingTopics(Map<PubSubTopic, Long> pubSubTopicsRetentions) {
+    List<PubSubTopic> topicsTobeCleanup = new ArrayList<>();
+    Set<PubSubTopic> kafkaTopics = apacheKafkaAdminAdapter.get().listAllTopics();
+    for (Map.Entry<PubSubTopic, Long> entry: pubSubTopicsRetentions.entrySet()) {
+      PubSubTopic pubSubTopic = entry.getKey();
+      if (!kafkaTopics.contains(pubSubTopic)) {
+        try {
+          String storeName = pubSubTopic.getStoreName();
+          String clusterDiscovered = admin.discoverCluster(storeName).getFirst();
+          Store store = admin.getStore(clusterDiscovered, storeName);
+          LOGGER.info("Find topic discrepancy case: {}", pubSubTopic);
```

Comment:
I'd suggest for these bad but not the end-of-world case, we use WARN. It's not good to have discrepancy among different pubsub ystems.